client/blocks/importer/wordpress/upgrade-plan/upgrade-plan-details.tsx

```diff
@@ -7,32 +7,36 @@ import { useI18n } from '@wordpress/react-i18n';
 import clsx from 'clsx';
 import React, { useState, useEffect } from 'react';
 import ButtonGroup from 'calypso/components/button-group';
-import QueryPlans from 'calypso/components/data/query-plans';
+import QuerySitePlans from 'calypso/components/data/query-site-plans';
 import { useSelectedPlanUpgradeMutation } from 'calypso/data/import-flow/use-selected-plan-upgrade';
 import { useSelector } from 'calypso/state';
-import { getCurrentUserCurrencyCode } from 'calypso/state/currency-code/selectors';
-import { getPlanRawPrice } from 'calypso/state/plans/selectors';
+import { getSitePlan, getSitePlanRawPrice } from 'calypso/state/sites/plans/selectors';
```

Comment:
It would be great if we started using the `usePricingMetaForGridPlans` hook for pricing (or the respective `usePlans` / `useSitePlans` queries) from the `@automattic/data-stores` package (`/packages/data-stores/plans`) - and stop using the Redux state.

We have work in progress for stripping out entirely the Calypso/Redux state for plans (and site plans) - see https://github.com/Automattic/wp-calypso/issues/86638

You should be able to reach the original or discounted price from `usePricingMetaForGridPlans` (I don't think there's any case that's not covered by that hook right now).

Any chance someone can follow up on this? :-)

@daledupreez

Reply:
Here's an example PR with similar refactor https://github.com/Automattic/wp-calypso/pull/89396 (passing in the `siteId` will generate prices equivalent to `getSitePlanRawPrice`

Reply:
@chriskmnds we can shift that across, yes. However, how should we handle the conflict that's effectively been flagged in https://github.com/Automattic/wp-calypso/issues/92210? The issue there is that we're not showing discounted prices because we're not showing renewal prices, and are relying on a separate notice for upgrade credits.

Also, this serves as preparatory work for us to take advantage of site-specific introductory offers on the Creator plan. Will those hooks surface the offers as we'd expect via the site plans API?

Reply:
It doesn't look like we can tell _why_ a discount was applied, which may be an issue -- we're not even getting the textual message back from the server via `usePricingMetaForGridPlans()` AFAICT.

Reply:
> The issue there is that we're not showing discounted prices because we're not showing renewal prices, and are relying on a separate notice for upgrade credits.

I am not sure. The hook does return "discountedPrice" and "originalPrice", which should give you a choice? See the [PlanPricing type](https://github.com/Automattic/wp-calypso/blob/trunk/packages/data-stores/src/plans/types.ts#L71) for details on what these are (TLDR the discounted price considering a `siteId` will be a prorated price, and without a `siteId` it will be a currency discount)

> Also, this serves as preparatory work for us to take advantage of site-specific introductory offers on the Creator plan. Will those hooks surface the offers as we'd expect via the site plans API?

Yes, they should. The same hook was used for the WOO intro offers and the WPCOM intro offers project later (if you recall). We map these to an `introOffer` property from the main query hook ([use-site-plans](https://github.com/Automattic/wp-calypso/blob/trunk/packages/data-stores/src/plans/queries/use-site-plans.ts)) and also resurface it from the `usePricingMetaForGridPlans` hook (so go with this first). The `usePlans`/`useSitePlans` hooks are closer to the API returns (just camelCased), while `usePricingMetaForGridPlans` is more for convenience. 

See the [PlanIntroductoryOffer](https://github.com/Automattic/wp-calypso/blob/trunk/packages/data-stores/src/plans/types.ts#L63) type:

```
export interface PlanIntroductoryOffer {
	formattedPrice: string;
	rawPrice: number;
	intervalUnit: string;
	intervalCount: number;
	isOfferComplete: boolean;
}
```

Reply:
and, of course, there's also a [useIntroOffers](https://github.com/Automattic/wp-calypso/blob/trunk/packages/data-stores/src/plans/hooks/use-intro-offers.ts) hook there. 

So I would first try to `usePricingMetaForGridPlans` to get the whole pricing data. If that won't work, reach for the individual `usePlans`/`useSitePlans` hooks, or any of the other selectors like `useIntroOffers`.

The idea is to not rely on Redux state. and clearly feel free to update/add any properties needed on these hooks

Reply:
> The issue there is that we're not showing discounted prices because we're not showing renewal prices, and are relying on a separate notice for upgrade credits.

The INR/PHP/MXN discount is a special type of discount that is not an introductory offer.  As a rule, we should not show any kind of discount if upgrade credits are present. However, the `usePricingMetaForGridPlans` does not know about upgrade credits, so it will return discounted prices if the INR/PHP/MXN discount is applicable, even if upgrade credits are present. So currently, it is the responsibility of the [PlanFeatures2023GridHeaderPrice component](https://github.com/Automattic/wp-calypso/blob/trunk/packages/plans-grid-next/src/components/header-price.tsx#L147-L151) to hide the discount if upgrade credits are present.

I recently worked on a PR that solves a similar issue here: https://github.com/Automattic/wp-calypso/pull/92205.

I wanted to highlight this note from the PR above:

> Note: When working on this PR, I realized that this bit of logic is critical to how plan prices are shown and probably should not be the responsibility of the components that display pricing. Going forward, if we add another way to display prices, the same logic would have to be repeated. IMO, this logic should be centralized and should be a part of the usePricingMetaForGridPlans hook. If upgrade credits are available, this hook should always return discounted prices as null.
We can ship this PR for now as a temporary fix and then iterate on adding this logic to usePricingMetaForGridPlans or wherever it is most appropriate.

There are two ways in which we can fix this issue:
* Update `usePricingMetaForGridPlans` so that it is aware of upgrade credits and does not return discounted prices if upgrade credits are present.
* Replace the special INR/PHP/MXN discount with an introductory offer implementation. I have raised this point here: pbOQVh-4Wn-p2#comment-6011

Reply:
I haven't been following this issue above, but these are separate issues. The state selector above should be straight up replaceable by any of the data-store hooks, irrespective. 

We may also be conflating concepts with intro offers. I think @daledupreez is referring to intro offers as the customary concept (as done previously with 1 dollar, etc). These are mapped to the hooks, so there's really nothing that's not serviceable from data-store already onthat end.

Anw any scenario of fixing selectors is probably best done on the hooks, not the Redux state. Otherwise, i'd just have to repeat the work :-)

Reply:
I created a card for the Migration Flow Improvement project. We'll pick it after finishing the work on introductory offer.
https://github.com/Automattic/wp-calypso/issues/92221

Reply:
Thanks. This feels a little more critical than to punt until after the intro offers work - especially if you'll be rendering introductory offers, the same framework/hooks should be used. 

We've done a bit of work/design  to enable that functionality through data-stores, so fresh implementations opting for previous/monolithic Redux state is certainly a step backward.

If y'all don't have capacity to make the refactor, then I will do it. Please let me know.

Also, I'd love it if we can agree that intro offers are served through the new hooks on the frontend.

@merkushin

Reply:
@chriskmnds 

Unfortunately, we can't take it in work immediately. My current work is considered critical for the project, so I can't put it on pause, and it uses the data we receive here, so changing the source of data might become a mess for one of us. It is probably better to delay the switch for one week. Do you think it is acceptable?

> Also, I'd love it if we can agree that intro offers are served through the new hooks on the frontend. Does this sound good?

[The initial approach](https://github.com/Automattic/wp-calypso/pull/92088/commits/682aa58c33dbe466d1ff1aa8f69224c8dd5d0042) was to use `useSitePlans` that had all needed data (site plans and introductory offers in them), but it had a side effect — repeated requests, described here: https://github.com/Automattic/wp-calypso/pull/92088#pullrequestreview-2141470047.

Reply:
> It is probably better to delay the switch for one week. Do you think it is acceptable?

Totally fine. I still have other/previous cases to refactor. My concern is this work (that also involves intro offers apparently) is that we build out more functionality over the Redux state that is already being treated as "deprecated". We are actively migrating things to the data-store hooks. We've already removed most plan pricing selectors and `getPlanDiscountedRawPrice` and `getPlanRawPrice` are next. Once these are migrated, then we will target the Query components `QueryPlans` and `QuerySitePlans` for removal. 

We also don't get many chances to target refactors like these. It's kinda by accident that now is the time that I can continue work on it, and I'm not sure for how long before another priority arises. We are generally treating refactors as secondary work.

> but it had a side effect — repeated requests, described here: https://github.com/Automattic/wp-calypso/pull/92088#pullrequestreview-2141470047

Was this a problem? To understand the issue, this wasn't loopy right (meaning, firing endlessly)? The React-query hooks can fire multiple requests in the background but that shouldn't be an issue to the frontend. Once the data settles (retrieved) then nothing will change unless the API data changes.

Reply:
p.s. all of the Signup flows that query plans data (including for introductory offers) are using the data-store hooks. So I'm not sure I see any blocker from using it from another Calypso component if that's what this work is.

The only case where we've had issues with the hooks was in ETK and things outside of Calypso - something unknown (yet) was not compiling the data properly there, which needs investigation. I don't think this is related though, but pls correct me if I'm wrong. Was the data retrieved and stabilized in your case (irrespective of multiple calls in the background)?

Reply:
@merkushin, me (or someone else from Serenity) can focus on getting this refactor done, ideally in parallel with your work to get the offers shown to users. That was my thinking all along, @chriskmnds -- we don't want to block this work, but we also don't want to let it linger in a bad state for very long.

Reply:
> It doesn't look like we can tell why a discount was applied, which may be an issue -- we're not even getting the textual message back from the server via usePricingMetaForGridPlans() AFAICT.

Hi, we have updated `useSitePlans`, so it returns an array of `cost_overrides`, which contains the reason for a discount. We also updated the `usePricingMetaForGridPlans` hook to include discounts from plan proration based on a parameter. I think you should now be able to use the `usePricingMetaForGridPlans` hook for your use case.

Reply:
Hi @aneeshd16!
Thank you! We're migrating to usePricingMetaForGridPlans here: https://github.com/Automattic/wp-calypso/pull/92201